### DISCUSSION TITLE
Contribute to Fief's localization 🌍

### DISCUSSION THREAD

**Discussion by frankie567** [fief-dev/fief#43] [Announcements]
We would like to provide Fief's **authentication pages** (login, registration, forgot password...) in lot of different languages.

You can help us by translating them to your native language! We use [POEditor](https://poeditor.com/) to help us manage the translations.

<p align="center">
    <a href="https://poeditor.com/join/project/1ViJq0Thro"><img src="https://md-buttons.francoisvoron.com/button.svg?text=Join%20the%20translation%20project%20on%20POEditor&bg=f43f5e&w=400&h=50&fs=18" alt="Join the translation project on POEditor"></a>
</p>

Here is the list of languages currently enabled in the project:

- [ ] Chinese (simplified) 🇨🇳
- [x] Taiwanese Mandarin 🇹🇼
- [x] English 🇬🇧
- [x] French 🇫🇷
- [ ] German 🇩🇪
- [ ] Italian 🇮🇹
- [ ] Portuguese (Portugal) 🇵🇹
- [x] Portuguese (Brazil) 🇧🇷
- [ ] Spanish 🇪🇸
- [ ] Polish 🇵🇱
- [ ] Russian 🇷🇺

*Legend*

- [x] *Complete*
- [ ] *Incomplete (help needed 🙏)*

---

If your language is not yet in the list, tell us in this thread so we can enable it on the project 👇

Thanks! Merci ! ¡ Gracias !

**Comment by danilopolani** [comment]
✅ Italian

**Reply by frankie567** [reply]
Wow, that's great! Thanks 🙏

**Comment by ChuckMoe** [comment]
:white_check_mark: German

**Reply by frankie567** [reply]
Thank you 🎉

**Comment by plpsanchez** [comment]
Portuguese

**Reply by plpsanchez** [reply]
Already did the Spanish.

**Reply by frankie567** [reply]
Thank you @plpsanchez 🙏  I've added the Portuguese language to the project!

**Reply by plpsanchez** [reply]
Great!  What main variation of Portuguese do you prefer?  I can do either or both of European-Portuguese and Brazilian-Portuguese, they are very very similar, but differ in the usual translation of a few needed words. For instance, the usual translation for screen in Portugal is "ecrã" (from the French écran) but in Brazilian portuguese the usual translation is "tela" (most people could not even guess what an ecrã could be).

**Reply by frankie567** [reply]
Well, let's say we go for European-Portuguese for now; we'll see in the future if we need the regional variations 😄

**Comment by su-network** [comment]
Hindi  ✨

**Comment by JimScope** [comment]
✅Spanish translation completed

**Reply by frankie567** [reply]
Thank you @JimScope, much appreciated 🙏

**Comment by michaldev** [comment]
Can you enable polish language? I'll translate it.

**Reply by frankie567** [reply]
Sure! Done 👍

**Reply by michaldev** [reply]
@frankie567  I noticed that there are also technical messages in the translations. Such as `prompt should either be "none", "login" or "register"` - do they need to be translated? In my country, people prefer technical messages in English.

**Reply by frankie567** [reply]
I'll trust you for this :) if you think it's better to keep them in English, you can just copy-paste them untouched.

**Reply by michaldev** [reply]
Done 👌

**Reply by michaldev** [reply]
@frankie567 Hi, Why I can't update polish translation? It is disabled in poeditor.

And I can't join to project: "This project can't accept new contributors at this time. Please try again later.".

**Reply by frankie567** [reply]
So, it seems we've hit POEditor free plan limit in terms of number of contributors. I'll ask them if we can have their Open Source plan.

**Reply by jamesnie4** [reply]
@frankie567  Hello, it seems that Russian is not localized yet. I want to contribute immediately, but I still cannot join "This project can't accept new contributors at this time. Please try again later."

**Comment by kiortir** [comment]
Can you add russian? I can give it a try

**Reply by frankie567** [reply]
Sure! Added 👍

**Comment by shirkan** [comment]
I can do Hebrew if you care, but usually Hebrew speakers will always use English

**Reply by frankie567** [reply]
Well, I'm always open to add new languages! I don't know much about Hebrew, but I think it's a right to left language, is it?

If so, this would need extra care on the CSS side to handle this correctly (never did this before 😅).

**Reply by shirkan** [reply]
yes It is, so much of a headache.

**Comment by csmaniottojr** [comment]
Portuguese (Brazil) completed :white_check_mark:

**Reply by frankie567** [reply]
Thank you @csmaniottojr 🙏

**Comment by ruipoliveira** [comment]
Portuguese completed!

**Reply by frankie567** [reply]
Obrigado 🙏 I've just updated the translations with new terms, if you have a moment to complete them 😊

**Comment by bingyulee** [comment]
Can you add Chinese (TW) zh-TW ? 
I'll give it a try.
Thanks.

**Reply by frankie567** [reply]
Hi @bingyulee 👋  Added!

**Reply by bingyulee** [reply]
Chinese (TW) zh-TW translated completed.

**Reply by bingyulee** [reply]
Hi @frankie567. I already have a self-hosted Docker service running and want to apply this translation within this service, what should I do?

**Reply by frankie567** [reply]
Well, currently nothing, you'll have to wait for a new release that integrates the new translations 😊

**Reply by bingyulee** [reply]
@frankie567  Thank you for your prompt response and the information. Any idea when the next release might be rolling out?  I understand that software development schedules can be fluid, so a rough estimate would be appreciated if possible. Or is there a way for me to package the Docker image with these translations myself? Thanks for all your hard work on this!

**Comment by LulzLoL231** [comment]
Hello!
Can I try to translate the project into Russian? As I see, there is not much work being done on it...

**Reply by frankie567** [reply]
Sure! Russian is already enabled on POEditor, so you can start right away: https://poeditor.com/projects/po_edit?id_language=134&per_page=20&id=508445

**Comment by jniemenmaa** [comment]
Swedish

**Comment by ysnbyzli** [comment]
Can you add Turkish? I can give it a try 🙏🏻
